csharp/generate/templates/ApiResponse.mustache

```diff
@@ -0,0 +1,178 @@
+{{>partial_header}}
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using RestSharp;
+
+namespace {{packageName}}.Client
+{
+    /// <summary>
+    /// Provides a non-generic contract for the ApiResponse wrapper.
+    /// </summary>
+    public interface IApiResponse
+    {
+        /// <summary>
+        /// The data type of <see cref="Content"/>
+        /// </summary>
+        Type ResponseType { get; }
+
+        /// <summary>
+        /// The content of this response
+        /// </summary>
+        Object Content { get; }
+
+        /// <summary>
+        /// Gets or sets the status code (HTTP status code)
+        /// </summary>
+        /// <value>The status code.</value>
+        HttpStatusCode StatusCode { get; }
+
+        /// <summary>
+        /// Gets or sets the HTTP headers
+        /// </summary>
+        /// <value>HTTP headers</value>
+        Multimap<string, string> Headers { get; }
+
+        /// <summary>
+        /// Gets or sets any error text defined by the calling client.
+        /// </summary>
+        string ErrorText { get; set; }
+
+        /// <summary>
+        /// Gets or sets any cookies passed along on the response.
+        /// </summary>
+        List<Cookie> Cookies { get; set; }
+
+        /// <summary>
+        /// The raw content of this response
+        /// </summary>
+        string RawContent { get; }
+
+        /// <summary>
+        /// Status of the response. Indicates whether any internal errors have been thrown.
+        /// </summary>
+        ResponseStatus ResponseStatus { get; }
+
+        /// <summary>
+        /// Potential internal exception that might occur with the processing of the API call
+        /// </summary>
+        Exception InternalException { get; }
```

Comment:
What is the value of adding this to the `ApiResponse`?

Reply:
Getting the stack trace so we can pinpoint where an exception happened anywhere in the SDK code, so that we could quickly debug it.